lib/Backend/BackendUtils.cpp

```diff
@@ -187,9 +187,29 @@ runtime::RuntimeBundle::create(const IRFunction &F,
   }
   auto constantMaxSize = constantAllocator.getMaxMemoryUsage();
 
-  // Compute the offsets for Placeholders.
+  // Placeholders should be allocated in a order of Input|InputOutput|Output.
+  std::vector<Placeholder const *> inputOutputPlaceholders;
+  std::vector<Placeholder const *> outputPlaceholders;
+  std::vector<Placeholder const *> contiguousPlaceholders;
   for (auto &v : F.findPlaceholders()) {
-    // Get the WeightVar for each Placeholder to calculate offsets.
+    if (isInput(v)) {
```

Comment:
This could be simpler if you do the inputOutput check first:

```
if (input && output) {
  inputOutputPlacholders.push_back(v);
} else if (input) {
  contiguousPlaceholders.push_back(v);
} else { 
  outputPlaceholders.push_back(v);
}
```

Reply:
> This could be simpler if you do the inputOutput check first:
> 
> ```
> if (input && output) {
>   inputOutputPlacholders.push_back(v);
> } else if (input) {
>   contiguousPlaceholders.push_back(v);
> } else { 
>   outputPlaceholders.push_back(v);
> }
> ```

Sure